src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -131,13 +131,23 @@ public object VisitSqlRoot(SqlRootExpression expression, SearchOptions context)
                 StringBuilder.AppendLine();
             }
 
-            StringBuilder.Append("FROM ").Append(VLatest.Resource).Append(" ").AppendLine(resourceTableAlias);
+            StringBuilder.Append("FROM ").Append(VLatest.Resource).Append(" ").Append(resourceTableAlias);
 
             if (expression.TableExpressions.Count > 0)
             {
-                StringBuilder.Append("INNER JOIN ").AppendLine(TableExpressionName(_tableExpressionCounter));
+                StringBuilder.AppendLine().Append("INNER JOIN ").AppendLine(TableExpressionName(_tableExpressionCounter));
                 StringBuilder.Append("ON ").Append(VLatest.Resource.ResourceSurrogateId, resourceTableAlias).Append(" = ").Append(TableExpressionName(_tableExpressionCounter)).AppendLine(".Sid1");
             }
+            else if (!_isHistorySearch)
+            {
+                // If not joining with any other CTEs and this is not a history search, make sure the optimizer does not decide to do a scan on the clustered index
+                StringBuilder.AppendLine(" WITH(INDEX(IX_Resource_ResourceTypeId_ResourceSurrgateId))");
```

Comment:
>IX_Resource_ResourceTypeId_ResourceSurrgateId [](start = 54, length = 45)

This feels like shooting bazooka to kill a mosquito... There should be a reason why SQL server pick clustered index over this one. Any chance we can have access to server on which this occurs or emulate that? I would love to see statistics and index fragmentation. Maybe that index is so fragmented what it doesn't yield much different time in execution.

Reply:
You can repro it on a small database too. My hypothesis is that it deems the clustered index preferable so that it does not have to do a key lookup to retrieve the raw resource column and others.